File: Ambulance.BLL/Models/PatientModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AmbulanceSystem.BLL.Models;

public class PatientModel
{
    public int PatientId { get; set; }
    public int PersonId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string SecondName { get; set; } = string.Empty;
    public List<string> Allergies { get; set; } = new List<string>();
    public List<string> ChronicDiseases { get; set; } = new List<string>();
    public List<CallModel> Calls { get; set; } = new List<CallModel>();
    public List<MedicalCardModel> MedicalCards { get; set; } = new List<MedicalCardModel>();
}

File: ambulance-site/src/api/models/PersonModels/PatientModel.js
import { MedicalCardModel } from "./MedicalCardModel.js";
import { CallModel } from "./CallModel.js";

export class PatientModel {
    constructor({
        personId = -1,
        allergies = [],
        chronicDiseases = [],
        calls = [],
        medicalCard = null,
    } = {}) {
        this.personId = personId;
        this.allergies = allergies; // залишаємо як є, бо це просто масив рядків
        this.chronicDiseases = chronicDiseases; // залишаємо як є, бо це просто масив рядків
        this.calls = calls.map(r => new CallModel(r));
        this.medicalCard = medicalCard ? new MedicalCardModel(medicalCard) : null;
    }
}